firstvoices/backend/serializers/import_job_serializers.py

```diff
@@ -0,0 +1,136 @@
+    CreateSiteContentSerializerMixin,
+    SiteContentLinkedTitleSerializer,
+)
+from backend.serializers.media_serializers import FileUploadSerializer
+from backend.serializers.utils import (
+    check_required_headers,
+    get_site_from_context,
+    validate_headers,
+)
+from backend.serializers.validators import SupportedFileType
+
+
+class ImportReportRowSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ImportJobReportRow
+        fields = ["row_number", "status", "errors"]
+
+    row_number = serializers.IntegerField(read_only=True)
+    status = serializers.CharField(read_only=True)
+    errors = serializers.ListField(child=serializers.CharField(), read_only=True)
+
+
+class ImportReportSerializer(serializers.ModelSerializer):
+    total_rows = serializers.IntegerField(read_only=True)
+    diff_headers = serializers.ListField(child=serializers.CharField(), read_only=True)
+    rows = ImportReportRowSerializer(many=True, read_only=True)
+    totals = serializers.JSONField(read_only=True)
+
+    class Meta:
+        model = ImportJobReport
+        fields = ["total_rows", "diff_headers", "totals"]
+
+
+class ImportJobSerializer(
+    CreateSiteContentSerializerMixin, SiteContentLinkedTitleSerializer
+):
+    id = serializers.UUIDField(read_only=True)
+    data = FileUploadSerializer(
+        validators=[SupportedFileType(mimetypes=["text/csv", "text/plain"])],
+    )
+    run_as_user = serializers.CharField(required=False)
+    validation_result = ImportReportSerializer(read_only=True)
+
+    class Meta:
+        model = ImportJob
+        fields = SiteContentLinkedTitleSerializer.Meta.fields + (
+            "mode",
+            "validation_result",
+            "run_as_user",
+            "data",
+        )
+
+    def create_file(self, file_data, filetype, site):
+        user = self.context["request"].user
+        file = filetype(
+            content=file_data,
+            site=site,
+            created_by=user,
+            last_modified_by=user,
+        )
+        file.save()
+        return file
+
+    def create(self, validated_data):
+        validated_data["site"] = get_site_from_context(self)
+        file = self.create_file(validated_data["data"], File, validated_data["site"])
+
+        try:
+            table = tablib.Dataset().load(
+                file.content.read().decode("utf-8-sig"), format="csv"
+            )
+
+            # Validate headers
+            # If required headers not present, raise ValidationError
+            check_required_headers(table.headers)
+
+            # else, print warnings for extra or invalid headers
+            validate_headers(table.headers)
+
+            # If the file is valid, create an ImportJob instance and save the file
+            title = validated_data.get("title", "")
+            mode = validated_data.get("mode", None)
+            run_as_user = validated_data.get("run_as_user", None)
+
+            entry = ImportJob(
```

Comment:
A question for @sarahfirstvoices as well since I am also working on an API the queues jobs. Should the `ImportJob` created here have a `task_id` as per thee `BaseJob` class?

Reply:
Good idea-- yes I think it would make life easier if we all use the base job serializer